### ISSUE TITLE
Add supported HotKeys to documentation

### ISSUE THREAD

**Issue by reubster** [DaxStudio/DaxStudio#828]
I would like to have a quick reference indicating which HotKey commands are available in DaxStudio. 

A new help article could be created with a table of the keys and a description. 

It looks like this would be as simple as documenting the mappings in /src/DaxStudio.UI/ViewModels/OptionsViewModel.cs in /docs/_documentation/features)/query-editor.md?

**Comment by marcosqlbi** [comment]
They can be configured in Options / Key Bindings, do you mean you would like to include the defaults in the documentation?

<img width="518" alt="image" src="https://user-images.githubusercontent.com/13840816/163669353-50eb48af-06e5-4b3f-ba2f-f962332022c3.png">

**Comment by reubster** [comment]
Yes, I've been using DaxStudio for years and didn't know you could do that. It's not described in the Options Window doc. 

https://daxstudio.org/documentation/features/options-window/

Happy to try doing it myself?

**Comment by marcosqlbi** [comment]
Good point! @dgosbell we could mention the Key Bindings page in the documentation! :)

**Comment by reubster** [comment]
Is there a plan / desire to move to Power Bi key mappings in the long run? Would seem intuitive to users?

**Comment by marcosqlbi** [comment]
Where do you see differences?
I think that the current implementation doesn't manage multiple keys in a sequence, which are used by Power BI in some cases.
However, the shortcuts used by DAX Studio are pretty standard, for this reason I'm asking.

**Comment by reubster** [comment]
I use VSCode a lot and the key sequences are a massive improvement. The main ones I need/ use are ALT+SHIFT up/down to duplicate a line… and CTRL+D to select the next instance of a term - super useful for renaming things.

**Comment by reubster** [comment]
The multi-cursor stuff is the ideal though. Once you get the hang of it it massively improves coding speed

**Comment by dgosbell** [comment]
So I can definitely add the Key Bindings to the docs, they were in preview for a long time and then I forgot to loop back to update the docs when we released them.

> Is there a plan / desire to move to Power Bi key mappings in the long run? Would seem intuitive to users?

Yes it makes sense to align with Power BI Desktop where possible, but I'm not aware of any keybindings that conflict with Power BI at the moment. Have you come across some in DAX Studio that are different to Power BI?

> The main ones I need/ use are ALT+SHIFT up/down to duplicate a line… and CTRL+D to select the next instance of a term - super useful for renaming things.

ALT + SHIFT + Up/Down is not possible as this combination is hard coded into the editor control for use as multi-line vertical editing and this matches with Notepad++ , Visual Studio and SSMS

I've never used the Ctrl+D combination in VS Code - how does it differ from just highlighting some text and using Ctrl+F ?

**Comment by reubster** [comment]
> how does it differ from just highlighting some text and using Ctrl+F ?

CTRL+D is a multi-_select_ command, not only finding, so "find the next instance and include it in the current cursor selection".

**Comment by dgosbell** [comment]
> CTRL+D is a multi-select command, not only finding, so "find the next instance and include it in the current cursor selection".

Oh OK, but I don't think I can support that since I do not believe the underlying edit control that I'm using in DAX Studio supports multi-caret selections. But while the multi-caret typing looks cool it also seems that you can do a similar thing with the standard find/replace.

**Comment by reubster** [comment]
Hiya, thanks for the response. Not quite sure I understand the point about multi-caret. Isn’t ALT+SH up/down a multi-caret action? 

In VSCode and Power Bi - it’s CTRL+ALT up/down for vertical editing. 

Purely my opinion… but I think VSCode/Aqua data studio… supersedes all of notepad++, Visual Studio and SSMS. 

I guess you’re supporting SSAS and AAS, though, not just Power Bi, so fair enough. 

Dax Studio is, nevertheless, the best app ever!

**Comment by dgosbell** [comment]
> Hiya, thanks for the response. Not quite sure I understand the point about multi-caret. Isn’t ALT+SH up/down a multi-caret action?

Yes, but it's a specific type of action and I have not implemented this in DAX Studio, it's a hotkey built in to the edit control. So I cannot intercept or change it as it is handled before the DAX Studio code runs. So while altering this is not impossible, it is a non-trivial amount of work.

> Purely my opinion… but I think VSCode/Aqua data studio… supersedes all of notepad++, Visual Studio and SSMS.

Yes, but VS Code is the newest of all these applications. All the others including DAX Studio are older than VS Code. And you are right that we also support SSAS and AAS. I'm happy to add other hotkey functions and to make them configurable through the options so that people can set them up they way they like, but there are some limitations like the ALT+SH+Arrows which are difficult to alter.

**Comment by dgosbell** [comment]
closing this as we have the hotkeys now in the docs at http://daxstudio.org/docs/features/options/options-window-key-bindings/
